salt/_modules/metalk8s_network.py

```diff
@@ -257,3 +257,24 @@ def get_control_plane_ingress_endpoint():
     return "https://{}:8443".format(
         __salt__["metalk8s_network.get_control_plane_ingress_ip"]()
     )
+
+
+def get_portmap_ips(as_cidr=False):
+    portmap_ips = set()
+    portmap_cidrs = __salt__["pillar.get"]("networks:portmap:cidr")
+
+    if portmap_cidrs:
+        if not isinstance(portmap_cidrs, list):
+            portmap_cidrs = [portmap_cidrs]
+        for cidr in portmap_cidrs:
+            portmap_ips = portmap_ips.union(
+                set(__salt__["network.ip_addrs"](cidr=cidr))
+            )
```

Comment:
Sets are mutable objects, for each logical operation on them (such as `union`, `intersection` or `difference`), there's an `_update` counterpart (in the case of `union`, the counterpart is just `update`, because it's quite easy to understand as is).

```suggestion
            portmap_ips.update(
                __salt__["network.ip_addrs"](cidr=cidr)
            )
```

Reply:
OH well, TIL, there is update for set, great I will change this thank you :+1: